core/src/main/scala/org/elasticmq/actor/queue/QueueActorMessageOps.scala

```diff
@@ -103,6 +103,15 @@ trait QueueActorMessageOps extends Logging {
         messageQueue += internalMessage
         None
       } else if (messagesById.contains(id.id)) {
+        // Putting the msg to dead letters queue if exists or delete.
+        if (internalMessage.receiveCount >= queueData.maxReceiveCount) {
+          logger.info(s"send message $internalMessage to dead letters actor $deadLettersActorRef")
+          deadLettersActorRef
+            .map(_ ! SendMessage(internalMessage.toNewMessageData))
+            .getOrElse(internalMessage.deliveryReceipt.foreach(dr => deleteMessage(DeliveryReceipt(dr))))
+          return None
```

Comment:
[minor] maybe instead of the `return`, this could be an if-else. That way the `if` is an expression whose result is returned. Also this method is quite long - maybe a method could be extracted, e.g. `receiveMessage(id)`?